tests/test_integration.py

```diff
@@ -12,6 +13,9 @@ def test_devpi_upload(devpi_server, uploadable_dist):
         devpi_server.password,
         str(uploadable_dist),
     ]
+
+    # Patching validate url to return https for devpiserver url
+    monkeypatch.setattr(utils, "validate_url", lambda repository_url: "https")
```

Comment:
This seems like a smell (that we're monkey-patching something that doesn't leave the computer, i.e., is not a network call)

Reply:
So we use `http://` URLs in [devpi-server](https://github.com/pypa/twine/blob/d5c6a98b01d4521c2ac5516fa7be61a95710b3ee/tests/conftest.py#L70) and [pypi-server](https://github.com/pypa/twine/blob/master/tests/conftest.py#L169) . 

If I leave these URLs as is with `http://`, the `validate_url` function fails since we are only expecting `https` schemes there and will fail [here](https://github.com/pypa/twine/blob/master/twine/utils.py#L115)

But if I change them to `https://`, making devpi-server and pypi-server support https is non-trivial and honestly more work than setting up the test needs. (e.g. see [Supporting HTTPS for pypiserver](https://pypi.org/project/pypiserver/#supporting-https)

So the best approach I could think of is to monkey-patch `validate_url` to return `https`, which will cause the `validate_url` to not fail, and the tests will pass.

Please let me know if there are better ways to resolve this, and I will look into implementing them.